--- a/TMessagesProj/src/main/java/org/telegram/ui/Cells/DialogCell.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/Cells/DialogCell.java
@@ -5523,14 +5523,14 @@
                 }
                 sb.append(". ");
             } else if (chat != null) {
+                sb.append(chat.title);
+                sb.append(". ");
                 if (chat.broadcast) {
                     sb.append(getString(R.string.AccDescrChannel));
                 } else {
                     sb.append(getString(R.string.AccDescrGroup));
                 }
                 sb.append(". ");
-                sb.append(chat.title);
-                sb.append(". ");
             }
         }
         if (drawVerified) {
