--- a/TMessagesProj/src/main/java/org/telegram/messenger/MediaController.java
+++ b/TMessagesProj/src/main/java/org/telegram/messenger/MediaController.java
@@ -136,6 +136,7 @@
 public class MediaController implements AudioManager.OnAudioFocusChangeListener, NotificationCenter.NotificationCenterDelegate, SensorEventListener {
 
     private native int startRecord(String path, int sampleRate);
+    private native void setVoiceRecordQuality(int quality); // 0=low, 1=medium, 2=high
 
     private native int writeFrame(ByteBuffer frame, int len);
 
@@ -4670,6 +4671,7 @@
                             return super.delete();
                         }
                     };
+                    setVoiceRecordQuality(MessagesController.getGlobalMainSettings().getInt("voiceRecordQuality", 2));
                     if (startRecord(recordingAudioFile.getPath(), sampleRate) == 0) {
                         AndroidUtilities.runOnUIThread(() -> {
                             recordStartRunnable = null;
@@ -4750,6 +4752,7 @@
             }
             AutoDeleteMediaTask.lockFile(recordingAudioFile);
             try {
+                setVoiceRecordQuality(MessagesController.getGlobalMainSettings().getInt("voiceRecordQuality", 2));
                 if (startRecord(recordingAudioFile.getPath(), sampleRate) == 0) {
                     AndroidUtilities.runOnUIThread(() -> {
                         recordStartRunnable = null;
